overlord/snapstate/handlers.go

```diff
@@ -315,19 +315,27 @@ func (m *SnapManager) installOneBaseOrRequired(t *state.Task, snapName string, c
 		return updatePrereqIfOutdated(t, snapName, contentAttrs, userID, flags)
 	}
 
+	// not installed, nor queued for install -> install it
+	deviceCtx, err := DeviceCtx(st, t, nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Creation of deviceCtx could be moved to inside the "else if" condition

Reply:
We discussed how deviceCtx is used later in the function.